display/display.go

```diff
@@ -246,6 +252,89 @@ func Init(version, commit, builtBy string) {
 		// Other potential keys are Tab and Backtab, they are ignored
 	})
 
+	searchBar.SetDoneFunc(func(key tcell.Key) {
+		tab := curTab
+
+		reset := func() {
+			searchBar.SetLabel("")
+			tabs[tab].applyAll()
+			App.SetFocus(tabs[tab].view)
+			layout.RemoveItem(searchBar)
+			tabs[tab].mode = tabModeDone
+		}
+
+		//nolint:exhaustive
+		switch key {
+		case tcell.KeyEnter:
+			// Figure out whether it's a URL, link number, or search
+			// And send out a request
+
+			// Escape the search string to not find regexp symbols
+			searchString = regexp.QuoteMeta(searchBar.GetText())
+
+			if strings.TrimSpace(searchString) == "" {
+				// Ignore
+				reset()
+				return
+			}
+
+			if tabs[tab].mode != tabModeSearch {
+				originalText = tabs[curTab].view.GetBytes(false)
+			}
+			tabs[tab].mode = tabModeSearch
+
+			// find all positions of the search string
+			searchRegex := regexp.MustCompile(searchString)
+			searchIdx := searchRegex.FindAllIndex(originalText, -1)
+
+			// find all positions of tags
+			tagsRegex := regexp.MustCompile(`\[.*?[^\[]\]`)
```

Comment:
This regex should be changed as I noted in my previous comment, and also it can be compiled once globally, instead of in this function.